File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void pre(TreeNode* root, int mx, int &ans){
        if(root == NULL) return ;
        if(root->val >= mx) {
            ans++;
            mx = root->val;
        }
        pre(root->left, mx, ans);
        pre(root->right, mx, ans);
    }
    int goodNodes(TreeNode* root) {
        int ans = 0;
        pre(root, INT_MIN, ans);
        return ans;
    }
};

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int cnt = 0;
    public int goodNodes(TreeNode root) {
        if(root == null) return cnt;
        dfs(root, root.val);
        return cnt;
    }
    private void dfs(TreeNode root, int max){
        if(root == null) return;
        if(root.val >= max) cnt++;
        dfs(root.left, Math.max(max, root.val));
        dfs(root.right, Math.max(max, root.val));
    }
}